algorithms/linfa-reduction/src/random_projection/gaussian/algorithms.rs

```diff
@@ -0,0 +1,62 @@
+use linfa::{prelude::Records, traits::Fit, Float};
+use ndarray::{Array, Array2, Ix2};
+use ndarray_rand::{
+    rand_distr::{Normal, StandardNormal},
+    RandomExt,
+};
+use rand::{prelude::Distribution, rngs::SmallRng, Rng};
+
+use super::super::common::johnson_lindenstrauss_min_dim;
+use super::hyperparams::GaussianRandomProjectionParamsInner;
+use super::{GaussianRandomProjectionParams, GaussianRandomProjectionValidParams};
+use crate::{impl_proj, ReductionError};
+
+/// Embedding via Gaussian random projection
+pub struct GaussianRandomProjection<F: Float> {
+    projection: Array2<F>,
+}
+
+impl<F, Rec, T, R> Fit<Rec, T, ReductionError> for GaussianRandomProjectionValidParams<R>
+where
+    F: Float,
+    Rec: Records<Elem = F>,
+    R: Rng + Clone,
+    StandardNormal: Distribution<F>,
+{
+    type Object = GaussianRandomProjection<F>;
+
+    fn fit(&self, dataset: &linfa::DatasetBase<Rec, T>) -> Result<Self::Object, ReductionError> {
```

Comment:
Also add some tests at least to check dimension reduction, it could be in the documentation like in [scikit-learn](https://scikit-learn.org/stable/modules/random_projection.html#gaussian-random-projection).

Reply:
Added in [a5f1ad3](https://github.com/rust-ml/linfa/pull/332/commits/a5f1ad3957775e90c9a898e7ce2d1ff02f26653e).